pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -3853,84 +3865,92 @@ func TestReconciler(t *testing.T) {
+						if err := ctrl.SetControllerReference(&tc.job, &tc.workloads[i], kClient.Scheme()); err != nil {
+							t.Fatalf("Could not setup owner reference in Workloads: %v", err)
+						}
+					}
+					tc.workloads[i].ResourceVersion = ""
+					if err := kClient.Create(ctx, &tc.workloads[i]); err != nil {
+						t.Fatalf("Could not create workload: %v", err)
 					}
 				}
-				if err := kClient.Create(ctx, &tc.workloads[i]); err != nil {
-					t.Fatalf("Could not create workload: %v", err)
-				}
-			}
-			recorder := &utiltesting.EventRecorder{}
-			reconciler := NewReconciler(kClient, recorder, append(tc.reconcilerOptions, jobframework.WithClock(t, fakeClock))...)
+				recorder := &utiltesting.EventRecorder{}
+				reconciler := NewReconciler(kClient, recorder, append(tc.reconcilerOptions, jobframework.WithClock(t, fakeClock))...)
 
-			jobKey := client.ObjectKeyFromObject(&tc.job)
-			_, err := reconciler.Reconcile(ctx, reconcile.Request{
-				NamespacedName: jobKey,
-			})
-			if diff := cmp.Diff(tc.wantErr, err, cmpopts.EquateErrors()); diff != "" {
-				t.Errorf("Reconcile returned error (-want,+got):\n%s", diff)
-			}
+				jobKey := client.ObjectKeyFromObject(&tc.job)
+				_, err := reconciler.Reconcile(ctx, reconcile.Request{
+					NamespacedName: jobKey,
+				})
+				if diff := cmp.Diff(tc.wantErr, err, cmpopts.EquateErrors()); diff != "" {
+					t.Errorf("Reconcile returned error (-want,+got):\n%s", diff)
+				}
 
-			var gotJob batchv1.Job
-			if err := kClient.Get(ctx, jobKey, &gotJob); client.IgnoreNotFound(err) != nil {
-				t.Fatalf("Could not get Job after reconcile: %v", err)
-			}
-			if diff := cmp.Diff(tc.wantJob, gotJob, jobCmpOpts...); diff != "" {
-				t.Errorf("Job after reconcile (-want,+got):\n%s", diff)
-			}
-			var gotWorkloads kueue.WorkloadList
-			if err := kClient.List(ctx, &gotWorkloads); err != nil {
-				t.Fatalf("Could not get Workloads after reconcile: %v", err)
-			}
+				var gotJob batchv1.Job
+				if err := kClient.Get(ctx, jobKey, &gotJob); client.IgnoreNotFound(err) != nil {
+					t.Fatalf("Could not get Job after reconcile: %v", err)
+				}
+				if diff := cmp.Diff(tc.wantJob, gotJob, jobCmpOpts...); diff != "" {
+					t.Errorf("Job after reconcile (-want,+got):\n%s", diff)
+				}
+				var gotWorkloads kueue.WorkloadList
+				if err := kClient.List(ctx, &gotWorkloads); err != nil {
+					t.Fatalf("Could not get Workloads after reconcile: %v", err)
+				}
 
-			wlCheckOpts := workloadCmpOpts
-			if useesPrebuiltWorkload {
-				wlCheckOpts = workloadCmpOptsWithOwner
-			}
+				wlCheckOpts := workloadCmpOpts
+				if useesPrebuiltWorkload {
+					wlCheckOpts = workloadCmpOptsWithOwner
+				}
 
-			if diff := cmp.Diff(tc.wantWorkloads, gotWorkloads.Items, wlCheckOpts...); diff != "" {
-				t.Errorf("Workloads after reconcile (-want,+got):\n%s", diff)
-			}
+				if features.Enabled(features.WorkloadRequestUseMergePatch) && tc.removeAdmissionAsInRealCluster {
+					tc.wantWorkloads[0].Status.Admission = nil
```

Comment:
I’m wondering if we can just ignore this field when the feature is disabled, since this is a bug with Apply Patch.